Library/Homebrew/test/dev-cmd/extract_spec.rb

```diff
@@ -53,7 +66,17 @@
         .and not_to_output.to_stderr
         .and be_a_success
       expect(path).to exist
-      expect(Formulary.factory(path).version).to be == "0.2"
+      expect(Formulary.factory(path).version).to be == "0.3"
+    end
+
+    it "retrieves the deprecated-syntax version of formula", :integration_test do
```

Comment:
I think it'd be good to only have a single `integration_test` in this file (as they are super slow)

Reply:
Fish out of water here. Should I remove `:integration_test` from all except `"retrieves the specified version of formula"`? As I understand it, that's probably the most-used `extract` variant.

Reply:
I think it would be ideal if the one integration test covered as many of the cases individually as possible. So, maybe run `brew extract --version=0.1` so it tests both checking out a specific version _and_ ignoring the old syntax issues

Reply:
Ah OK, thanks @Rylan12, will conflate as suggested.

While we're on the subject, what's the effective difference between a test case tagged with `:integration_test`, and one without?

Reply:
The `brew` method called in (most if not all) integration tests spins up a complete `brew` instance (i.e. by shelling out to `brew.rb`) instead of just calling the methods being tested directly. The downside is that this adds a lot of time. It's essentially like running `brew extract` (or whatever) directly from the command line instead of calling a method from within an already-up ruby instance.

I know that `:integration_test` needs to be specified in order to use the `brew` method, but it's possible that specifying it also causes some extra things to be done in preparation for the test that might also slow it down. Not 100% sure about that, though.

Reply:
Yup, I agree with everything @Rylan12 said! Sorry for being unclear here @gromgit.